node/src/accountant/payment_adjuster/mod.rs

```diff
@@ -0,0 +1,2340 @@
+            "DEBUG: {test_name}: \n\
+|Payable Account                            Balance Wei
+|
+|                                           Original
+|                                           Adjusted
+|
+|0x0000000000000000000000000000000000616263 100,000,000,000,000,000,000
+|                                           90,000,000,000,000,000,000
+|
+|Ruled Out Accounts                         Original
+|
+|0x0000000000000000000000000000000000646566 500,000,000,000,000,000,000
+|0x0000000000000000000000000000000000676869 250,000,000,000,000,000,000"
+        );
+        TestLogHandler::new().exists_log_containing(&log_msg.replace("|", ""));
+        test_inner_was_reset_to_null(subject)
+    }
+
+    #[test]
+    fn late_error_after_transaction_fee_adjustment_but_rechecked_transaction_fee_found_fatally_insufficient(
+    ) {
+        init_test_logging();
+        let test_name = "late_error_after_transaction_fee_adjustment_but_rechecked_transaction_fee_found_fatally_insufficient";
+        let now = SystemTime::now();
+        let balance_1 = multiple_by_billion(500_000_000_000);
+        let account_1 =
+            make_plucked_qualified_account("abc", balance_1, 300_000_000_000, 100_000_000_000);
+        // This account is eliminated in the transaction fee cut
+        let balance_2 = multiple_by_billion(111_000_000_000);
+        let account_2 =
+            make_plucked_qualified_account("def", balance_2, 50_000_000_000, 10_000_000_000);
+        let balance_3 = multiple_by_billion(300_000_000_000);
+        let account_3 =
+            make_plucked_qualified_account("ghi", balance_3, 150_000_000_000, 50_000_000_000);
+        let mut subject = PaymentAdjusterReal::new();
+        subject.logger = Logger::new(test_name);
+        let disqualification_arbiter = DisqualificationArbiter::default();
+        let disqualification_limit_2 =
+            disqualification_arbiter.calculate_disqualification_edge(&account_2);
+        // This is exactly the amount which will provoke an error
+        let cw_service_fee_balance_minor = disqualification_limit_2 - 1;
+        let qualified_payables = vec![account_1, account_2, account_3];
+        let analyzed_payables = convert_collection(qualified_payables);
+        let agent = BlockchainAgentMock::default()
+            .service_fee_balance_minor_result(cw_service_fee_balance_minor);
+        let adjustment_setup = PreparedAdjustment {
+            agent: Box::new(agent),
+            adjustment_analysis: AdjustmentAnalysis::new(
+                Adjustment::TransactionFeeInPriority {
+                    affordable_transaction_count: 2,
+                },
+                analyzed_payables,
+            ),
+            response_skeleton_opt: None,
+        };
+
+        let result = subject.adjust_payments(adjustment_setup, now);
+
+        let err = match result {
+            Ok(_) => panic!("expected an error but got Ok()"),
+            Err(e) => e,
+        };
```

Comment:
You should revisit the declaration of the agent, to make it use `debug`. It would allow us to use `unwrap_err()` here.